[Bug] Softplus激活函数缺少复数类型支持导致测试失败

### 问题描述

在运行 `test_activation_op.py` 中的softplus相关测试时，复数类型的测试用例出现失败。

### 相关测试代码

#### 测试类定义：

```python
class TestSoftplus(TestActivation):
    def setUp(self):
        self.op_type = "softplus"
        self.python_api = paddle.nn.functional.softplus
        self.init_dtype()
        self.init_shape()

        beta = 2
        threshold = 15

        np.random.seed(1024)
        x = np.random.uniform(-1, 1, self.shape).astype(self.dtype)
        if self.dtype == np.complex64 or self.dtype == np.complex128:
            x = (
                np.random.uniform(-1, 1, self.shape)
                + 1j * np.random.uniform(-1, 1, self.shape)
            ).astype(self.dtype)
        out = ref_softplus(x, beta, threshold)
        self.inputs = {'X': x}
        self.attrs = {'beta': beta, "threshold": threshold}
        self.outputs = {'Out': out}

    def init_shape(self):
        self.shape = [10, 12]

    def test_check_output(self):
        self.check_output(
            check_pir=True,
            check_pir_onednn=self.check_pir_onednn,
            check_symbol_infer=False,
        )

    def test_check_grad(self):
        if self.dtype == np.float16:
            return
        self.check_grad(
            ['X'], 'Out', check_pir=True, check_pir_onednn=self.check_pir_onednn
        )


class TestSoftplus_Complex64(TestSoftplus):
    def init_dtype(self):
        self.dtype = np.complex64

    def test_check_grad(self):
        self.check_grad(
            ['X'],
            'Out',
            max_relative_error=0.06,
            check_pir=True,
            check_pir_onednn=self.check_pir_onednn,
        )


class TestSoftplus_Complex128(TestSoftplus):
    def init_dtype(self):
        self.dtype = np.complex128
```

#### 参考实现：

```python
def ref_softplus(x, beta=1, threshold=20):
    x_beta = beta * x
    out = np.select(
        [x_beta <= threshold, x_beta > threshold],
        [np.log(1 + np.exp(x_beta)) / beta, x],
    )
    return out
```

### 详细错误信息

#### TestSoftplus_Complex128 错误：

**test_check_grad 失败：**
```
AssertionError: 2.100812924987562 not less than or equal to 0.005 : Operator softplus error, Gradient Check On Place(gpu:0) variable X (shape: (10, 12), dtype: complex128) max gradient diff 2.100813e+00 over limit 5.000000e-03, the first error element is 0, expected 9.530294e-03+1.925088e-02j, but got 8.129173e-03+1.518001e-03j.
```

**test_check_output 失败：**
```
AssertionError: 
Not equal to tolerance rtol=1e-05, atol=1e-05
Operator (softplus) Output (Out) has diff at Place(gpu:0) in static checker
Mismatched elements: 120 / 120 (100%)
Max absolute difference: 2.12846166
Max relative difference: 4.80560841
```

#### TestSoftplus_Complex64 错误：

**test_check_grad 失败：**
```
AssertionError: 2.1020372 not less than or equal to 0.06 : Operator softplus error, Gradient Check On Place(gpu:0) variable X (shape: (10, 12), dtype: complex64) max gradient diff 2.102037e+00 over limit 6.000000e-02, the first error element is 0, expected 9.530783e-03+1.925230e-02j, but got 8.129174e-03+1.518001e-03j.
```

**test_check_output 失败：**
```
AssertionError: 
Not equal to tolerance rtol=1e-05, atol=1e-05
Operator (softplus) Output (Out) has diff at Place(gpu:0) in static checker
Mismatched elements: 120 / 120 (100%)
Max absolute difference: 2.1284618
Max relative difference: 4.805607
```

### 根本原因

**缺少复数类型特化实现**：Paddle的 `SoftplusFunctor` 和 `CudaSoftplusFunctor` 类只有实数版本，没有针对复数类型 `ComplexType<T>` 的特化实现。

当输入复数时，Paddle会使用实数版本的实现，这导致：
1. 复数运算被错误处理
2. 结果与NumPy参考实现不匹配
3. 产生巨大的数值差异

### 影响范围

- `TestSoftplus_Complex64::test_check_output`
- `TestSoftplus_Complex64::test_check_grad`
- `TestSoftplus_Complex128::test_check_output`
- `TestSoftplus_Complex128::test_check_grad`

### 解决方案

需要为softplus函数实现复数类型特化：

1. **添加CPU复数特化**：`SoftplusFunctor<ComplexType<T>>`
2. **添加GPU复数特化**：`CudaSoftplusFunctor<ComplexType<T>>`
3. **确保正确的复数数学处理**

--- a/paddle/phi/kernels/funcs/activation_functor.h
+++ b/paddle/phi/kernels/funcs/activation_functor.h
@@ -860,6 +860,27 @@ struct SoftplusFunctor : public BaseActivationFunctor<T> {
   }
 };
 
+template <typename T>
+struct SoftplusFunctor<ComplexType<T>>
+    : public BaseActivationFunctor<ComplexType<T>> {
+  float beta;
+  float threshold;
+
+  typename BaseActivationFunctor<ComplexType<T>>::AttrPair GetAttrs() {
+    return {{"beta", &beta}, {"threshold", &threshold}};
+  }
+
+  template <typename Device, typename X, typename Out>
+  void operator()(Device d, X x, Out out) const {
+    auto x_beta = static_cast<ComplexType<T>>(beta) * x;
+    out.device(d) =
+        (x_beta > static_cast<ComplexType<T>>(threshold))
+            .select(x,
+                    (static_cast<ComplexType<T>>(1) + x_beta.exp()).log() /
+                        static_cast<ComplexType<T>>(beta));
+  }
+};
+
 // For numerical stability, using the following formula instead of
 // d(softplus(x))/dx = 1 / (1 + exp(-x))
 // d(softplus(x))/dx = 1 / (1 + exp(-beta * x)) when beta * x <= threshold(beta
@@ -4225,6 +4246,30 @@ struct CudaSoftplusFunctor : public BaseActivationFunctor<T> {
     MPType t = static_cast<MPType>(threshold);
     MPType x_beta = x * static_cast<MPType>(beta);
     return static_cast<T>(x_beta > t ? x : log(one + exp(x_beta)) / b);
+  }
+};
+
+template <typename T>
+struct CudaSoftplusFunctor<ComplexType<T>>
+    : public BaseActivationFunctor<ComplexType<T>> {
+  using MPType = typename phi::dtype::MPTypeTrait<ComplexType<T>>::Type;
+  MPType one = static_cast<MPType>(1.0f);
+  float beta;
+  float threshold;
+
+  typename BaseActivationFunctor<ComplexType<T>>::AttrPair GetAttrs() {
+    return {{"beta", &beta}, {"threshold", &threshold}};
+  }
+
+  // softplus(x) = beta * x > threshold ? x : log(1 + exp(beta * x)) / beta
+  __device__ __forceinline__ ComplexType<T> operator()(
+      const ComplexType<T> arg_x) const {
+    MPType x = static_cast<MPType>(arg_x);
+    MPType b = static_cast<MPType>(beta);
+    MPType t = static_cast<MPType>(threshold);
+    MPType x_beta = x * static_cast<MPType>(beta);
+    return static_cast<ComplexType<T>>(x_beta > t ? x
+                                                  : log(one + exp(x_beta)) / b);
   }
 };
 

